test/ClientTest.php

```diff
@@ -39,6 +39,8 @@ class ClientTest extends TestCase
     /** @var Client */
     protected $xmlrpcClient;
 
+    private MockObject $mockedIntrospector;
```

Comment:
It looks like this is only referenced in a public method `mockIntrospector()` that appears unused… Is this test case extended elsewhere or is there a load of dead code to remove here?

Reply:
I used php 8.3 locally, so it show deprecated of dynamic property, the method is filling property that will call expect() later.

Since it is in test, mark private should be fine.

Using MockObject is fine imo, since it will just works, eg when check on phpstan. Intersection is usually used for direct implements both, eg: `implements \Iterator, \Countable` => ` \Iterator&\Countable`